File: swift/src/DataTypes.swift
import Foundation

// Project: DataTypes
// Practice: Variables, optionals, type inference

let personName = "Judah"
let personAge = 25
let personEmail: String? = nil

print("=== Swift Data Types Demo ===")
print("Name: \(personName)")
print("Age: \(personAge)")
print("Email: \(personEmail ?? "Not provided")")

// Type inference
var score = 88.5                  // Inferred as Double
let isActive = true              // Inferred as Bool

print("\nScore: \(score)")
print("Active: \(isActive)")

// Collections
let fruits = ["Apple", "Banana", "Cherry"]
var numbers = [1, 2, 3, 4, 5]

print("\nFruits: \(fruits)")
print("Numbers: \(numbers)")

numbers.append(6)
print("After append: \(numbers)")

// Dictionary
let person = ["name": "Alice", "city": "San Francisco", "job": "Engineer"]
print("\nPerson info:")
for (key, value) in person {
    print("  \(key): \(value)")
}

File: kotlin/src/DataTypes.kt
// Project: DataTypes
// Practice: Variables, nullable types, type inference

fun main() {
    println("=== Kotlin Data Types Demo ===")
    
    val personName = "Judah"
    val personAge = 25
    val personEmail: String? = null

    val person2name = "judah"
    val person2age = 21
    val personalemail2: String? = null
    
    println("Name: $personName")
    println("Age: $personAge")
    println("Email: ${personEmail ?: "Not provided"}")
    
    // Type inference
    var score = 88.5              // Inferred as Double
    val isActive = true           // Inferred as Boolean
    
    println("\nScore: $score")
    println("Active: $isActive")
    
    // Collections (immutable by default)
    val fruits = listOf("Apple", "Banana", "Cherry")
    val numbers = mutableListOf(1, 2, 3, 4, 5)
    
    println("\nFruits: $fruits")
    println("Numbers: $numbers")
    
    numbers.add(6)
    println("After add: $numbers")
    
    // Map
    val person = mapOf(
        "name" to "Alice",
        "city" to "San Francisco",
        "job" to "Engineer"
    )
    
    println("\nPerson info:")
    for ((key, value) in person) {
        println("  $key: $value")
    }
    
    // Nullable types and smart casting
    val value: Int? = 42
    if (value != null) {
        println("\nValue is not null: $value")
    }
}
